### DISCUSSION TITLE
help with dupe bot

### DISCUSSION THREAD

**Discussion by psmpm** [PrismarineJS/mineflayer#1878] [Q&A]
i'm writing some code for a dupe server where the bot will automatically eat an enchanted golden apple if it gets too hungry and dupe it if the count gets below 32. however, when I run my code, the command never gets run. can anyone help?
```js
bot.on("health", async () => {
    if (bot.health < 20) {
        bot.quit();
        console.log("damage taken");
        process.exit(0);
    }
    console.log("hungry");
    bot.setQuickBarSlot(8)
    if (bot.heldItem.count < 32) {
        while (bot.heldItem.count < 32) {
            setTimeout(() => {
                bot.chat("/dupe");
            },250);
        }
    }
    if (bot.food <= 16) {
        if (bot.heldItem.name !== "enchanted_golden_apple") {
            console.log("slot 9 of hotbar was not enchanted_golden_apple");
            process.exit(0);
        } else {
            await bot.consume();
        }
    }
});
```

**Comment by u9g** [comment]
Have you tried debugging this? ie... putting break points to see if the health event is being called, if bot.health is changing, etc...

**Comment by IceTank** [comment]
Your while loop runs forever. So setTimeout has no chance off handeling the bot.chat command.
